### ISSUE TITLE
Search Service 400 Error - The provided value for encryptionWithCmk.enforcement were invalid

### ISSUE THREAD

**Issue by mitcheman** [hashicorp/terraform-provider-azurerm#26609]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.9.2

### AzureRM Provider Version

3.111

### Affected Resource(s)/Data Source(s)

azurerm_search_service

### Terraform Configuration Files

```hcl
resource "azurerm_search_service" "main" {
  name                = var.name
  resource_group_name = var.resource_group_name
  location            = var.resource_group_location
  sku                 = "basic"

  tags = var.tags
}
```

### Debug Output/Panic Output

```shell
Search Service Name: "search_service_name"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: : The provided value for encryptionWithCmk.enforcement were invalid.
│ Parameter name: encryptionWithCmk
```

### Expected Behaviour

I have not provided any input for the "customer_managed_key_enforcement_enabled" argument so it should default to False to update my Search Service as needed. The Search Service already exists and is just updating tags correctly. 

### Actual Behaviour

Receiving a 400 Error with the error: 

"performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: : The provided value for encryptionWithCmk.enforcement were invalid.
│ Parameter name: encryptionWithCmk"

### Steps to Reproduce

Run a Terraform Apply against an existing search service.

**Comment by liuwuliuyun** [comment]
Hi @mitcheman, I appreciate you bringing this to our attention. I attempted to replicate the issue with the AzureRM provider version 3.111, as you recommended, but it worked as expected. Below is my entire template and command line output. Could you please share your complete template so I can try to reproduce the issue on my end?

Reproduce template I use
```
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.111.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "test" {
  name     = "testyunliu1"
  location = "East US"
}

resource "azurerm_search_service" "testsearch" {
  name                = "testsearchyunliu1"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
  sku                 = "basic"
  tags = {
    environment = "test"
  }
}
```
Command Line Output of Above template
```
PS C:\Users\yunliu1\Documents\terraformtest\search> terraform init

Initializing the backend...

Initializing provider plugins...
- Finding hashicorp/azurerm versions matching "3.111.0"...
- Installing hashicorp/azurerm v3.111.0...
- Installed hashicorp/azurerm v3.111.0 (signed by HashiCorp)

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
PS C:\Users\yunliu1\Documents\terraformtest\search> terraform plan

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_resource_group.test will be created
  + resource "azurerm_resource_group" "test" {
      + id       = (known after apply)
      + location = "eastus"
      + name     = "testyunliu1"
    }

  # azurerm_search_service.testsearch will be created
  + resource "azurerm_search_service" "testsearch" {
      + customer_managed_key_enforcement_enabled = false
      + hosting_mode                             = "default"
      + id                                       = (known after apply)
      + local_authentication_enabled             = true
      + location                                 = "eastus"
      + name                                     = "testsearchyunliu1"
      + partition_count                          = 1
      + primary_key                              = (sensitive value)
      + public_network_access_enabled            = true
      + query_keys                               = (known after apply)
      + replica_count                            = 1
      + resource_group_name                      = "testyunliu1"
      + secondary_key                            = (sensitive value)
      + sku                                      = "basic"
      + tags                                     = {
          + "environment" = "test"
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 

Note: You didn't use the -out option to save this plan, so Terraform can't guarantee to take exactly these actions if you run "terraform apply" now.
PS C:\Users\yunliu1\Documents\terraformtest\search> terraform apply -auto-approve

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_resource_group.test will be created
  + resource "azurerm_resource_group" "test" {
      + id       = (known after apply)
      + location = "eastus"
      + name     = "testyunliu1"
    }

  # azurerm_search_service.testsearch will be created
  + resource "azurerm_search_service" "testsearch" {
      + customer_managed_key_enforcement_enabled = false
      + hosting_mode                             = "default"
      + id                                       = (known after apply)
      + local_authentication_enabled             = true
      + location                                 = "eastus"
      + name                                     = "testsearchyunliu1"
      + partition_count                          = 1
      + primary_key                              = (sensitive value)
      + public_network_access_enabled            = true
      + query_keys                               = (known after apply)
      + replica_count                            = 1
      + resource_group_name                      = "testyunliu1"
      + secondary_key                            = (sensitive value)
      + sku                                      = "basic"
      + tags                                     = {
          + "environment" = "test"
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.
azurerm_resource_group.test: Creating...
azurerm_resource_group.test: Still creating... [10s elapsed]
azurerm_resource_group.test: Creation complete after 12s [id=/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/testyunliu1]
azurerm_search_service.testsearch: Creating...
azurerm_search_service.testsearch: Still creating... [10s elapsed]
azurerm_search_service.testsearch: Still creating... [20s elapsed]
azurerm_search_service.testsearch: Creation complete after 23s [id=/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/testyunliu1/providers/Microsoft.Search/searchServices/testsearchyunliu1]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

**Comment by mitcheman** [comment]
I was actually able to fix this manually with the Azure REST API. I believe at some point azure or some service changed my Azure Search Service as having the following config in its ARM template:

`        "encryptionWithCmk": {
            "enforcement": null,
            "encryptionComplianceStatus": null
        },`
        
 As I don't believe null is a valid value for enforcement it was erroring out on every Azure Search resource change. Even the CLI wasn't working. 
 
I was able to fix this by using a PUT method on the management API. First getting the configuration, saving it to a file, editing the encryptionWIthCmk settings and running the PUT request. I attempted to use PATCH but that didn't seem to work.

`az rest --method GET \
  --uri "https://management.azure.com/subscriptions/subID/resourceGroups/azRGName/providers/Microsoft.Search/searchServices/searchServiceName?api-version=2021-04-01-preview" \
  > search_service_definition.json`
  
  
  `az rest --method PUT \
  --uri "https://management.azure.com/subscriptions/subID/resourceGroups/azRGName/providers/Microsoft.Search/searchServices/searchServiceName?api-version=2021-04-01-preview" \
  --headers "Content-Type=application/json" \
  --body @search_service_definition.json`
